jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy

```diff
@@ -0,0 +1,267 @@
+
+                dir('cortx-re') {
+                    checkout([$class: 'GitSCM', branches: [[name: '*/main']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', depth: 1, honorRefspec: true, noTags: true, reference: '', shallow: true], [$class: 'AuthorInChangelog']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: 'https://github.com/Seagate/cortx-re']]])
+                }
+
+                // Install tools required for release process
+                 sh label: 'Installed Dependecies', script: '''
+                    yum install -y expect rpm-sign rng-tools python3-pip
+                    ln -fs $(which python3.6) /usr/bin/python2
+                    systemctl start rngd
+                '''
+
+                // Integrate components rpms
+                sh label: 'Collect Release Artifacts', script: '''
+                    
+                    pushd ${COMPONENTS_RPM}
+                        for component in `ls -1 | grep -E -v "${COMPONENT_NAME}"`
+                        do
+                            echo -e "Copying RPM's for $component"
+                            if ls $component/last_successful/*.rpm 1> /dev/null 2>&1; then
+                                cp $component/last_successful/*.rpm "${CORTX_ISO_LOCATION}"
+                            fi
+                        done
+                    popd
+
+                    # Symlink 3rdparty repo artifacts
+                    ln -s "${THIRD_PARTY_DEPS}" "${THIRD_PARTY_LOCATION}"
+                        
+                    # Symlink python dependencies
+                    ln -s "${PYTHON_DEPS}" "${PYTHON_LIB_LOCATION}"
+                '''
+
+                // sh label: 'RPM Signing', script: '''
+                //     pushd cortx-re/scripts/rpm-signing
+                //         cat gpgoptions >>  ~/.rpmmacros
+                //         sed -i 's/passphrase/'${PASSPHARASE}'/g' genkey-batch
+                //         gpg --batch --gen-key genkey-batch
+                //         gpg --export -a 'Seagate'  > RPM-GPG-KEY-Seagate
+                //         rpm --import RPM-GPG-KEY-Seagate
+                //     popd
+
+
+                //     pushd cortx-re/scripts/rpm-signing
+                //         chmod +x rpm-sign.sh
+                //         cp RPM-GPG-KEY-Seagate ${CORTX_ISO_LOCATION}
+                //         for rpm in `ls -1 ${CORTX_ISO_LOCATION}/*.rpm`
+                //         do
+                //             ./rpm-sign.sh ${PASSPHARASE} ${rpm}
+                //         done
+                //     popd
+
+                // '''
+                
+                sh label: 'RPM Signing', script: '''
+                    pushd ${CORTX_ISO_LOCATION}
+                        rpm -qi createrepo || yum install -y createrepo
+                        createrepo .
+                    popd
+                '''    
+
+                sh label: 'RPM Signing', script: '''
```

Comment:
Please update the label name as appropriate.

Reply:
done